src/packagedcode/opam.py

```diff
@@ -0,0 +1,309 @@
+
+    @property
+    def purl(self):
+        return PackageURL(
+                    type='opam',
+                    name=self.name
+                ).to_string()
+
+
+# Regex expressions to parse file lines
+parse_file_line = re.compile(
+    r'(?P<key>^(.+?))'
+    r'\:\s*'
+    r'(?P<value>(.*))'
+).match
+
+parse_dep = re.compile(
+    r'^\s*\"'
+    r'(?P<name>[A-z0-9\-]*)'
+    r'\"\s*'
+    r'(?P<version>(.*))'
+).match
+
+"""
+Example:
+>>> p = parse_file_line('authors: "BAP Team"')
+>>> assert p.group('key') == ('authors')
+>>> assert p.group('value') == ('"BAP Team"')
+
+>>> p = parse_dep('"bap-std" {= "1.0.0"}')
+>>> assert p.group('name') == ('bap-std')
+>>> assert p.group('version') == ('{= "1.0.0"}')
+"""
+
+def parse_opam(location):
+    """
+    Return a dictionary containing all the opam data.
+    """
+    with io.open(location, encoding='utf-8', closefd=True) as data:
+        lines = data.readlines()
+
+    opam_data = {}
+
+    for i, line in enumerate(lines):
+        parsed_line = parse_file_line(line)
+        if parsed_line:
+            key = parsed_line.group('key').strip()
+            value = parsed_line.group('value').strip()
+            if 'maintainer' == key:
+                stripped_val = value.strip('["] ')
+                stripped_val = stripped_val.split('" "')
+                opam_data[key] = stripped_val
+                continue
+            if 'authors' == key:
+                if '[' in line: # If authors are present in multiple lines
+                    for authors in lines[i+1:]:
+                        value += ' ' + authors.strip()
+                        if ']' in authors:
+                            break
+                    value = value.strip('["] ')
+                else:
+                    value = get_stripped_data(value)   
+                value = value.split('" "')
+                opam_data[key] = value
+                continue
+            if 'depends' == key: # Get multiline dependencies
+                value = []
+                for dep in lines[i+1:]:
+                    if ']' in dep:
+                        break
+                    parsed_dep = parse_dep(dep)
+                    if parsed_dep:
+                        value.append(Opam(
+                                name=parsed_dep.group('name').strip(),
+                                version=parsed_dep.group('version').strip('{ } ').replace('"', '')
+                            )
+                        )
+                opam_data[key] = value
+                continue
+            if 'description' == key: # Get multiline description
+                value = ''
+                for cont in lines[i+1:]:
+                    value += ' ' + cont.strip()
+                    if '"""' in cont:
+                        break
+
+            opam_data[key] = get_stripped_data(value)
+
+    return opam_data
+
+
+def get_stripped_data(data):
+    """
+    Return data after removing unnecessary special character.
```

Comment:
Do you have specific tests for this? and may rather than `get_stripped_data` would it be best called `clean_data`

Reply:
We haven't made any specific test here. Maybe we don't need any specific test here.

Reply:
having unit tests for each function is always a better idea.

Reply:
I have added tests for this now :)